tracer/src/Datadog.Trace/Telemetry/Collectors/ConfigurationTelemetry.Collector.cs

```diff
@@ -68,36 +79,64 @@ public void CopyTo(IConfigurationTelemetry destination)
                 return data;
             }
 
-            while (config.TryDequeue(out var entry))
-            {
-                data.Add(GetConfigKeyValue(entry));
-            }
+            GetData(config, data);
 
             return data;
+        }
 
-            static ConfigurationKeyValue GetConfigKeyValue(ConfigurationTelemetryEntry entry)
+        private static void GetData(ConcurrentQueue<ConfigurationTelemetryEntry> buffer, List<ConfigurationKeyValue> destination)
+        {
+            while (buffer.TryDequeue(out var entry))
             {
-                return new ConfigurationKeyValue(
-                    name: entry.Key,
-                    origin: entry.Origin.ToStringFast(),
-                    seqId: entry.SeqId,
-                    error: entry.Error,
-                    value: GetValue(entry));
-            }
+                destination.Add(GetConfigKeyValue(entry));
 
-            static object? GetValue(ConfigurationTelemetryEntry entry)
-            {
-                return entry.Type switch
+                if (ConfigurationKeysMapping.TryGetValue(entry.Key, out var mappedKey))
                 {
-                    ConfigurationTelemetry.ConfigurationTelemetryEntryType.Bool => entry.BoolValue,
-                    ConfigurationTelemetry.ConfigurationTelemetryEntryType.Double => entry.DoubleValue,
-                    ConfigurationTelemetry.ConfigurationTelemetryEntryType.Int => entry.IntValue,
-                    ConfigurationTelemetry.ConfigurationTelemetryEntryType.Redacted => "<redacted>",
-                    _ => entry.StringValue
-                };
+                    destination.Add(MapConfigKeyValueAsString(entry, mappedKey));
```

Comment:
Hmmm... I wonder if this could be an issue. The _order_ we create `ConfigurationTelemetryEntry`s is important, because they contain a `SeqId` that's incremented every time they're created, and it's built-in to telemetry that subsequent `SeqId`s overwrite previous ones 🤔 And these keys are going to be mapped to the existing values in the intake.

Anyway, I _think_ it should be ok because we have a queue, and dequeuing is FIFO. So when we create these, it'll overwrite the "original" version (including subsequent values of the original), but then _those_ will also get overwritten later.

So I think it's ok, unless I've missed something? But are we _100%_ sure we need these duplicates at all, because if not, the problem goes away for sure.

Reply:
I think that's ok because the mapped keys are created in the same order